src/m365/planner/commands/roster/roster-remove.spec.ts

```diff
@@ -0,0 +1,152 @@
+    auth.service.connected = true;
+    auth.service.accessTokens[(command as any).resource] = {
+      accessToken: 'abc',
+      expiresOn: new Date()
+    };
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    promptOptions = undefined;
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+    auth.service.accessTokens = {};
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ROSTER_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('prompts before removing the specified roster when confirm option not passed with id', async () => {
+    await command.action(logger, {
+      options: {
+        id: validRosterId
+      }
+    });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified plan when confirm option not passed and prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
+    await command.action(logger, {
+      options: {
+        id: validRosterId
+      }
+    });
+    assert(deleteSpy.notCalled);
+  });
+
+  it('correctly deletes plan by id', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/planner/rosters/${validRosterId}`) {
+        return;
+      }
+
+      throw 'Invalid Request';
+    });
+
+    await command.action(logger, {
+      options: {
+        verbose: true,
+        id: validRosterId,
+        confirm: true
+      }
+    });
+  });
+
+  it('correctly deletes plan by id when prompt confirmed', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/planner/rosters/${validRosterId}`) {
+        return;
+      }
+
+      throw 'Invalid Request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        id: validRosterId
+      }
+    });
+  });
+
+  it('correctly handles random API error', async () => {
+    sinon.stub(request, 'delete').callsFake(() => Promise.reject('The requested item is not found.'));
```

Comment:
Instead of throwing a plain text error, let's throw the entire error object to mimic the API as good as possible.